src/webapp/pages/dashboard/mock.tsx

```diff
@@ -0,0 +1,107 @@
+import i18n from "$/utils/i18n";
+import { TableColumn } from "@eyeseetea/d2-ui-components";
+import { Progress } from "$/webapp/components/progress/Progress";
+
+type ReferenceObject<T> = {
+    id: string;
+    selectable?: boolean | undefined;
+} & T;
+
+export type Step = {
+    position: number;
+    status: "default" | "complete";
+};
+
+export interface TableModel {
+    id: string;
+    name: string;
+    dataset: string;
+    startDate: number;
+    endDate: number;
+    status: "In Progress" | "Completed";
+    progress: Step[];
+    lastModification: string;
+}
+
+export const columns: TableColumn<TableModel>[] = [
+    { name: "name", text: i18n.t("Name"), sortable: true },
+    { name: "dataset", text: i18n.t("Dataset"), sortable: true },
+    { name: "startDate", text: i18n.t("Start Date"), sortable: true },
+    { name: "endDate", text: i18n.t("End Date"), sortable: true },
+    { name: "status", text: i18n.t("Status"), sortable: true },
+    {
+        name: "progress",
+        text: i18n.t("Progress"),
+        sortable: false,
+        getValue: row => {
+            return <Progress value={row.progress} />;
```

Comment:
I know for now we only need the Progress component in this table, but we can make the component more reusable for a future by separating the implementation from the value of the row. Something like:

```tsx
// ProgressContainer is just a styled component
<ProgressContainer>
    { 
      row.progres.map((value) =>    
       // This is the implementation you already have in <ListElement />
      <ProgressStatus status={value.status} text={value.position} />)
    }    
</ProgressContainer>
```

now we can reuse the `<ProgressStatus />` component in any part of the app not just this table, imagine we need to display the status in every step, we can directly call this component:

```tsx
// steps/Step1.tsx
<p>Outlier:<p/> <ProgressStatus status="complete" text="2" />
```

does it make sense?

Reply:
Yes, it makes perfect sense. Done!